### ISSUE TITLE
interpolation within/outside quotes give different results

### ISSUE THREAD

**Issue by derrickburns** [typesafehub/config#231]
It appears that resolving

```json
a {
   x: "${HOME}"
}
```

is different than resolving:

```json
a {
  x: ${HOME}
}
````

In the former case, the variable is not resolved.  In the latter it is.  

This is problematic. 

Consider when calling parseMap. The result is a Config.  What happens if we call resolve() on that Config?  Will it resolve values or not?  I suspect that it will.  But what if we don't want certain values to be resolved and others to be resolved.  There is no way to distinguish the two cases.

In other words, the escape mechanism of using quotation marks to distinguish to the two cases doesn't offer any value for people reading from maps.  Instead, a different escape mechanism is needed.

**Comment by derrickburns** [comment]
For what it's worth, I think that further clarity is needed for the file format, the ADT, and the translation of a file into the ADT. 

For example, it seems perfectly reasonable to me to offer a resolve() method of Config (as you have) whose semantics have nothing to do with the source of the Config, whether it be memory, a HOCON file or a YAML file.  

But perhaps I don't understand enough about the problem.

**Comment by havocp** [comment]
resolve() doesn't parse any strings. HOCON ${} syntax parses into a special kind of node that resolve looks to resolve. There's no way to create such a node with parseMap. You can get one using parseString("${foo}"). It isn't the same as a string node.

**Comment by derrickburns** [comment]
I see. Thanks.
